### ISSUE TITLE
iPad crashes - LightGallery v1.1.5 and v1.2

### ISSUE THREAD

**Issue by maarcin** [sachinchoolur/lightGallery#152]
Hi,

I recently tested a v1.1.5 a lot with different settings. On pc everything works really good, no issues at all. On iPad gallery crashed right after opening it. I used your solution from demo page and mixed lightgallery with lightslider. That caused a crash. So I saw that there is v1.2 in the making which looks so awesome. I gave it a try with the iPad, because on pc, again, it works awesome. But right from the start there are crashes on iPad. I check those two demos with thumbnails: http://sachinchoolur.github.io/lightgallery1.2/demos/

Both crashes right after the gallery is opened. I checked the code and compared it to the one from v1.1.5 that I set up. The thing is when I mixes both scripts and used .on('jg.complete', function() to start lightgallery it crashes. When I use only lightgallery without this it seems to work just fine, even when there are 40+ images. I used lightslider + lightgallery, on your v1.2 demo site you're using justified gallery  + lightgallery. So maybe that's something that may cause the crashes.

Just to let you know I use justified gallery script with lightgallery v1.1.5 too on my site and it works just fine on iPad.

So to sum things up:
1. lightgallery v1.1.5 + lightslider + on('jg.complete', function () - crash
2. lightgallery v1.1.5 + justified + on('jg.complete', function () - works fine
3. lightgallery v1.2 (from demo site) + justified + on('jg.complete', function () - crash
4. lightgallery v1.2 (all other demos), lightgallery v1.1.5 (even 40+ photos) - works fine

Cheers

**Comment by sachinchoolur** [comment]
Hi,
Thank you very much for your detailed information. 
I don't have an ipad to test this issue. It seems like memory issue with the ipad. 
I had tested this issue on ipad earlier and it was happening when i used more than 100 images at a time.  
It was happening because of css transitions. I am using only position, scale, rotation and opacity properties for animations which browser can animate easily without any problem. Then i tried with css left property instead of translate and it worked fine. I have added `useLeft` setting for forcing lightgallery to use left transition instead of translate. 

Please check this issue once again with `useLeft: true` and let me know.

If still it is happening please try to use minimal settings (without thumbnail, autoplay.. etc).

**Comment by sachinchoolur** [comment]
Hi,
Please re-open the issue if it still exist...

Thanks

**Comment by maarcin** [comment]
Hi,

I tested based on the demo from this repository with 25 images. With <code>useLeft: true</code> and with minimal settings it still crashes.

Cheers

**Comment by carasmo** [comment]
I added this CSS last, when you have a large number of images can you test if it still crashes?

```
.lg-img-wrap * {
 /*CSS transitions*/
 -o-transition-property: none !important;
 -moz-transition-property: none !important;
 -ms-transition-property: none !important;
 -webkit-transition-property: none !important;
 transition-property: none !important;

 /*CSS transforms*/
 -o-transform: none !important;
 -moz-transform: none !important;
 -ms-transform: none !important;
 -webkit-transform: none !important;
 transform: none !important;

 /*CSS animations*/
 -webkit-animation: none !important;
 -moz-animation: none !important;
 -o-animation: none !important;
 -ms-animation: none !important;
 animation: none !important;
}
```

**Comment by sachinchoolur** [comment]
https://github.com/sachinchoolur/lightGallery/issues/188#issuecomment-150906371
